src/server/game_server.rs

```diff
@@ -126,24 +144,49 @@ impl GameServer {
         }
     }
     
-    pub fn on_new_attack(&mut self, attack: EntityAttack) {
+    pub fn on_new_attack(&mut self, attack: EntityAttack, player_id: usize) {
         let victim = attack.victim_id();
-        // Communicate the attack to the player !
+        match attack.entity_kind() {
```

Comment:
You will ned to refactor this function because you won't have access to `entity_kind`. But it should not be too hard.